ran-core/src/main/java/io/ran/ClazzMethod.java

```diff
@@ -83,12 +92,61 @@ public boolean isSynchronized() {
 		return Modifier.isSynchronized(method.getModifiers());
 	}
 
+	private List<Type> getAllGenericTypes() {
+		ArrayList<Type> types = new ArrayList<>();
+		types.add(method.getGenericReturnType());
+		types.addAll(Arrays.asList(method.getGenericParameterTypes()));
+		types.addAll(Arrays.asList(method.getGenericExceptionTypes()));
+		return types.stream().flatMap(this::flattenGenericTypes).collect(Collectors.toList());
```

Comment:
Should this maybe be a set? Since a generic type can appear in multiple places but the caller of the method has no way to differentiate them in the return

Reply:
done